compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -1038,14 +1040,26 @@ private STToken processIdentifierOrKeyword() {
             case LexerTerminals.JOIN:
                 return getSyntaxToken(SyntaxKind.JOIN_KEYWORD);
             case LexerTerminals.RE:
-                return getSyntaxToken(SyntaxKind.RE_KEYWORD);
+                if (isNextTokenBacktick()) {
+                    return getSyntaxToken(SyntaxKind.RE_KEYWORD);
+                }
             default:
 //                if (this.keywordModes.contains(KeywordMode.QUERY)) {
 //                    return getQueryCtxKeywordOrIdentifier(tokenText);
 //                }
                 return getIdentifierToken();
         }
     }
+    
+    private boolean isNextTokenBacktick() {
+        int lookaheadCount = 0;
```

Comment:
Shouldn't this start with 1?

Reply:
No, reader.peek(0) will refers to next character after `e` right?